src/codegen/llvm.rs

```diff
@@ -112,6 +134,31 @@ impl LLVMValue for ir::Instruction {
                 s.push_str(")");
                 s
             }
+            Malloca { id, typ } => format!(
+                "%{id} = call i8* @malloc(i64 {type_size})",
+                id = id.borrow(),
+                type_size = typ.size()
+            ),
+            BitCast {
+                id,
+                from_id,
+                target_type,
+            } => format!(
+                "%{id} = bitcast i8* %{from} to {target_type}",
+                id = id.borrow(),
+                from = from_id.borrow(),
+                target_type = target_type.llvm_represent()
+            ),
+            Store {
+                source,
+                destination,
+            } => format!(
+                "store {} {}, {} %{}",
+                source.type_().llvm_represent(),
+                source.llvm_represent(),
+                (ir::Type::Pointer(source.type_().into())).llvm_represent(),
+                destination.borrow()
```

Comment:
Here can be better, once we replacing all `ID` in instructions with `Variable` concept. Because a variable should contain type information. The problem here is generated llvm would always correct, even target is not matching source type.